src/projections/tmerc.cpp

```diff
@@ -259,42 +259,31 @@ static PJ *setup_approx(PJ *P) {
 /*****************************************************************************/
 
 /* Helper functios for "exact" transverse mercator */
-#ifdef _GNU_SOURCE
-    inline
-#endif
-static double gatg(double *p1, int len_p1, double B) {
-    double *p;
-    double h = 0, h1, h2 = 0, cos_2B;
+inline
+static double gatg(const double *p1, int len_p1, double B, double cos_2B, double sin_2B) {
+    double h = 0, h1, h2 = 0;
 
-    cos_2B = 2*cos(2*B);
-    p = p1 + len_p1;
+    const double two_cos_2B = 2*cos_2B;
+    const double* p = p1 + len_p1;
     h1 = *--p;
     while (p - p1) {
-        h = -h2 + cos_2B*h1 + *--p;
+        h = -h2 + two_cos_2B*h1 + *--p;
         h2 = h1;
         h1 = h;
     }
-    return (B + h*sin(2*B));
+    return (B + h*sin_2B);
 }
 
 /* Complex Clenshaw summation */
-#ifdef _GNU_SOURCE
-    inline
-#endif
-static double clenS(double *a, int size, double arg_r, double arg_i, double *R, double *I) {
-    double      *p, r, i, hr, hr1, hr2, hi, hi1, hi2;
-    double      sin_arg_r, cos_arg_r, sinh_arg_i, cosh_arg_i;
+inline
+static double clenS(const double *a, int size,
+                    double sin_arg_r, double cos_arg_r,
+                    double sinh_arg_i, double cosh_arg_i,
+                    double *R, double *I) {
+    double      r, i, hr, hr1, hr2, hi, hi1, hi2;
 
     /* arguments */
-    p = a + size;
-#ifdef _GNU_SOURCE
-    sincos(arg_r, &sin_arg_r, &cos_arg_r);
```

Comment:
Any reason why you would removed `sincos`? Some instruction sets have `FSINCOS` (e.g. [here](https://www.felixcloutier.com/x86/fsincos)) that calculate the two simultaneously, often with fewer CPU cycles. It's also not just `_GNU_SOURCE`, as it's part of Intels math library too. There are other CXX initiatives like [FastTrigo](https://github.com/divideconcept/FastTrigo) to seek other optimizations.

Reply:
With gcc and a reasonable level of optimization, successive calls to sin and cos are converted to a call to sincos.  So, it's probably preferable to let the compiler handle this.

Reply:
Thanks, that's a reasonable assumption that I've confirmed using GCC with `-O1` (and up).